### PR TITLE
More specific path traversal check.

### PR DESCRIPTION
**Pull request by ndm2**
refs #7260

Totally forgot about it...

### CODE REVIEW

File: tests/TestCase/Network/ResponseTest.php [hunk 1]
```diff
@@ -1156,17 +1156,30 @@ public function testFileNotFound()
...
+    /**
+     * test file with ..\
+     *
+     * @expectedException \Cake\Network\Exception\NotFoundException
+     * @expectedExceptionMessage The requested file contains `..` and will not be read.
+     * @return void
+     */
+    public function testFileWithBackwardSlashPathTraversal() {
+        $response = new Response();
+        $response->file('my\..\cat.gif');
```
Comment (markstory) [inline tests/TestCase/Network/ResponseTest.php]: Wouldn't this have passed before as well?

Comment (ndm2) [inline tests/TestCase/Network/ResponseTest.php]: Yeah, I forgot the test that actually puts the changed behavior to a test :)

**Comment by ndm2** [comment]
The check could be made even more specific, so that stuff like `/path../` or `/.../` would be allowed too. It felt a little risky, so I didn't tried it for #7260, but it actually should be relatively simple, something like

```php
if (preg_match('#(^|(/|\\\\))\.\.(/|\\\\)#', $path) !== 0) {
```

Maybe that way this change would feel better, as the check would then really only target strings that resemble path traversal fragments (aside that `\..\` is possibly a valid name on *nix systems)?

**Comment by markstory** [comment]
I'm not sure that regex is 'simple'. The current code is fine. It covers the bad cases and doesn't block useful cases.

**Pull request merged**